src/lib/linked-list.js

```diff
@@ -0,0 +1,77 @@
+'use strict';
+
+const Node = require('./node');
+
+module.exports = class LinkedList {
+  constructor() {
+    this.length = 0;
+    this.head = null;
+  }
+
+  // Big O = O(1)
+  insertAtHead(value) {
+    const node = new Node(value);
+
+    node.next = this.head;
+    this.head = node;
+    this.length += 1;
+    return this;
+  }
+
+  // Big O = O(n)
+  find(value) {
+    if (!this.head) {
+      throw new Error('__ERROR__ The list is empty');
+    }
+
+    if (this.head.value === value) {
+      return this.head;
+    }
+
+    let currentNode = this.head;
+    while (currentNode.next) {
+      if (currentNode.next.value === value) {
+        return currentNode.next;
+      }
+      currentNode = currentNode.next;
+    }
+    return null;
+  }
+
+  // Big O = O(n)
+  pop(position) {
```

Comment:
Good job with the pop method, but remember what "pop" means - it takes off the last item in a collection and returns it. That sad, is your `position` argument really necessary in this case if you know "pop" is only ever taking off the LAST item in the collection?